src/patches/mapkey.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <absl/strings/ascii.h>
+#include <key.h>
+#include <map>
+#include <modifierkey.h>
+#include <patches/gamefunctions.h>
+#include <prime/KeyCode.h>
+#include <string>
+
+class MapKey
+{
+private:
+  static std::map<GameFunction, MapKey*> mappedKeys;
```

Comment:
As this is just an enum. It would be better to just have a std::array here instead of using a map.
Similarly, Is there a particular reason that we are storing a raw pointer here? Looks to me like you could just store the object directly, storing a pointer will add an additional indirection, while minor in perf, still adds up eventually.

_Side note: Generally you don't need map an can use/should prefer using unordered_map_